components/input/Input.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component, createElement } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { IconChevronUpSmallOutline, IconChevronDownSmallOutline, IconErrorSmallOutline } from '@teamleader/ui-icons';
+import InputMetaPropTypes from './InputMetaPropTypes';
+import Counter from '../counter';
+import theme from './theme.css';
+
+export default class Input extends Component {
+  static propTypes = {
+    bold: PropTypes.bool,
+    className: PropTypes.string,
+    counter: PropTypes.number,
+    disabled: PropTypes.bool,
+    icon: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
+    iconPlacement: PropTypes.oneOf(['left', 'right']),
+    id: PropTypes.string,
+    meta: InputMetaPropTypes,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+    placeholder: PropTypes.string,
+    readOnly: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+    step: PropTypes.number,
+    type: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  };
+
+  static defaultProps = {
+    iconPlacement: 'left',
+    disabled: false,
+    placeholder: '',
+    readOnly: false,
+    size: 'medium',
+    step: 1,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.isNumberInput = Boolean(props.type === 'number');
```

Comment:
Not really a fan of "storing" this in `this`. Now it's actually not possible to change an input from a number input to a normal one (or vice versa). We'll probably never do that, but you kind of expect that it would work.

Reply:
Can you change this as well @driesd?

Reply:
Dropped the storage in `this.`.

 :ok: by @driesd on 13/02/2018 14:14:48